src/app/src/test/kotlin/org/imperial/mrc/hint/unit/controllers/ErrorReportControllerTests.kt

```diff
@@ -47,6 +58,47 @@ class ErrorReportControllerTests
         assertThat(result.statusCode).isEqualTo(HttpStatus.BAD_REQUEST)
     }
 
+    @Test
+    fun `can serialize request body`()
+    {
+        val client = mock<Client> {
+            onGeneric { executeRequest(any()) } doReturn
+                    Response(statusCode = 200, body = DefaultBody(), url = URL("http://a.com"))
+        }
+        FuelManager.instance.client = client
+
+        val sut = ErrorReportController(FuelFlowClient(ObjectMapper(),
+                ConfiguredAppProperties(readPropsFromTempFile("issue_report_url=http://flow.com"))))
+
+        sut.postErrorReport(data)
+
+        verify(client).executeRequest(check {
+
+            assertThat(it.url.toString()).isEqualTo("http://flow.com")
+
+            assertThat(it.header("Content-Type")).isEqualTo(listOf("application/json"))
+
+            val response = ObjectMapper().readValue<JsonNode>(it.body.asString("application/json"))
+
+            assertThat(response["email"].asText()).isEqualTo("test.user@example.com")
+            assertThat(response["country"].asText()).isEqualTo("Kenya")
+            assertThat(response["projectName"].asText()).isEqualTo("Kenya2022")
+            assertThat(response["section"].asText()).isEqualTo("Model")
+            assertThat(response["jobId"].asText()).isEqualTo("123")
+            assertThat(response["errors"][0]["key"].asText()).isEqualTo("fomot-hasah-livad")
+            assertThat(response["errors"][0]["error"].asText()).isEqualTo("test error msg")
+            assertThat(response["errors"][0]["detail"].asText()).isEqualTo("#65ae0d095ea")
+            assertThat(response["errors"][1]["key"].asText()).isEqualTo("fomot-hasah-livid")
+            assertThat(response["errors"][1]["error"].asText()).isEqualTo("test error msg2")
+            assertThat(response["errors"][1]["detail"].asText()).isEqualTo("#25ae0d095e1")
+            assertThat(response["description"].asText()).isEqualTo("")
+            assertThat(response["stepsToReproduce"].asText()).isEqualTo("test steps")
+            assertThat(response["browserAgent"].asText()).isEqualTo("test agent")
+            assertThat(Instant.ofEpochSecond(response["timeStamp"]["epochSecond"].asLong()))
+                    .isEqualTo(Instant.ofEpochSecond(timeStamp.epochSecond))
```

Comment:
Adding this test seems to have revealed a problem: we don't want to send a field of the form `timestamp: {epochSecond: ... }` to Flow. I assume it's expecting an ISO date format string - not something specific to Java. Safer to use `String` rather than `Instant` in `ErrorReport.kt`? That way it will just be passed through. Or you'll need to ensure that it's (de)serialized correctly.